libr/core/canal.c

```diff
@@ -1597,6 +1559,118 @@ static int core_anal_graph_nodes(RCore *core, RAnalFunction *fcn, int opts, PJ *
 			}
 			free (str);
 		}
+	}
```

Comment:
I suggest to split this into the smaller functions.

Reply:
Yeah, will refactor the code after it is completely functional